samcli/lib/deploy/deployer.py

```diff
@@ -403,9 +407,9 @@ def describe_stack_events(self, stack_name, time_stamp_marker, **kwargs):
 
     @staticmethod
     def _check_stack_complete(status: str) -> bool:
-        return "COMPLETE" in status and "CLEANUP" not in status
+        return "IN_PROGRESS" not in status
```

Comment:
This one is easy to miss but it's important. When `disable-rollback` was specified and the deployment failed, the process stayed hanging because `UPDATE_FAILED` or `CREATE_FAILED` don't have `COMPLETE` in them. I'm not sure why it would be considered that it's not complete or still in progress if it has already failed creating/updating, so I just updated this. Would there be any problem doing this (was there a reason why update/create failed were not considered complete)?